modules/Core/src/main/java/org/terasology/rendering/nui/layers/hud/InventoryGridHud.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2014 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.rendering.nui.layers.hud;
+
+import org.terasology.engine.Time;
+import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.input.BindButtonEvent;
+import org.terasology.input.MouseInput;
+import org.terasology.logic.characters.CharacterComponent;
+import org.terasology.logic.inventory.InventoryUtils;
+import org.terasology.logic.inventory.SelectedInventorySlotComponent;
+import org.terasology.logic.players.LocalPlayer;
+import org.terasology.registry.CoreRegistry;
+import org.terasology.registry.In;
+import org.terasology.rendering.nui.BaseInteractionListener;
+import org.terasology.rendering.nui.Canvas;
+import org.terasology.rendering.nui.InteractionListener;
+import org.terasology.rendering.nui.databinding.ReadOnlyBinding;
+import org.terasology.rendering.nui.events.NUIMouseClickEvent;
+import org.terasology.rendering.nui.layers.ingame.inventory.InventoryCell;
+import org.terasology.rendering.nui.layouts.relative.RelativeLayout;
+
+public class InventoryGridHud extends CoreHudWidget {
+
+    @In
+    private LocalPlayer localPlayer;
+
+    private InteractionListener interactionListener = new BaseInteractionListener() {
+        @Override
+        public boolean onMouseClick(NUIMouseClickEvent event) {
+            MouseInput mouseButton = event.getMouseButton();
+            if (mouseButton == MouseInput.MOUSE_LEFT) {
+                EntityRef movingItem = localPlayer.getCharacterEntity().getComponent(CharacterComponent.class).movingItem;
+                EntityRef item  = InventoryUtils.getItemAt(movingItem, 0);
+                if(!item.exists()) {
+                    return true;
+                }
+                localPlayer.activateOwnedEntityAsClient(item);
```

Comment:
actually usually a right click activates an item. The idea behind this whole drop item out of window feature is to provide the user a way to get rid of items that fill up the inventory. Since an activation does not necesarly does this, please change it to a item drop for all items.